File: 프로그래머스/연습문제/Lv1/둘만의 암호/둘만의 암호.cpp
#include <string>
#include <vector>
#include <map>
using namespace std;

string solution(string s, string skip, int index) {
    map<char, bool> in;
    for(int i=0; i<skip.length(); i++)
        in[skip[i]] = true;
    
    for(int i=0; i<s.length(); i++){
        for(int j=0; j<index; j++){
            s[i]++;
            if(s[i] > 'z') s[i] = 'a';
            
            while(in[s[i]]){
                s[i]++;
                if(s[i] > 'z') s[i] = 'a';
            }
        }
    }
    
    return s;
}

File: 프로그래머스/연습문제/Lv1/둘만의 암호/둘만의 암호.py
def solution(s, skip, index):
    answer = ""
    for i in range(len(s)):
        ch = s[i]
        for j in range(index):
            ch = chr(ord(ch)+1)
            if ch > 'z': ch = 'a'

            while ch in skip:
                ch = chr(ord(ch)+1)
                if ch > 'z': ch = 'a'
        answer += ch

    return answer
